### PR TITLE
fix running indicator by ignoring PSRL aborts

### PR DESCRIPTION
**Pull request by TylerLeonhardt**
Fixes https://github.com/PowerShell/vscode-powershell/issues/2284

PSReadLine being cancelled was causing any operation that would show the running indicator to be immediately cancelled because a second message was being sent with `ExecutionStatus.Aborted`.

This ignores that one from PSRL.

### CODE REVIEW

File: src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs [hunk 1]
```diff
@@ -1838,9 +1838,15 @@ public MinifiedRunspaceDetails(RunspaceDetails eventArgs)
         /// <param name="e">details of the execution status change</param>
         private void PowerShellContext_ExecutionStatusChangedAsync(object sender, ExecutionStatusChangedEventArgs e)
         {
-            _languageServer?.SendNotification(
-                "powerShell/executionStatusChanged",
-                e);
+            // The cancelling of the prompt (PSReadLine) causes an ExecutionStatus.Aborted to be sent after every
+            // actual execution (ExecutionStatus.Running) on the pipeline. We ignore that event since it's counterintuitive to
+            // the goal of this method which is to send updates when the pipeline is actually running something.
```
Comment (TylerLeonhardt) [inline src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs]:
```suggestion
            // the goal of this method which is to send updates when the pipeline is actually running something.
            // In the event that we don't know if it was a ReadLine cancel, we default to sending the notification.
```

File: src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs [hunk 2]
```diff
@@ -1838,9 +1838,16 @@ public MinifiedRunspaceDetails(RunspaceDetails eventArgs)
...
+            // In the event that we don't know if it was a ReadLine cancel, we default to sending the notification.
+            if (!e?.ExecutionOptions?.IsReadLine ?? true)
```
Comment (SeeminglyScience) [inline src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs]: It's really hard to tell what this is doing, maybe change to something like this:

```csharp
var options = e?.ExecutionOptions;
if (options == null || !options.IsReadLine)
```

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs]: but I like question marks 😄

will change 👍

Comment (TylerLeonhardt) [reply src/PowerShellEditorServices/Services/PowerShellContext/PowerShellContextService.cs]: I used your logic

**Pull request merged**